tests/components/blebox/test_cover.py

```diff
@@ -0,0 +1,379 @@
+
+
+class TestShutter(CoverTest):
+    """Tests for cover devices representing a BleBox ShutterBox."""
+
+    def default_mock(self):
+        """Return a default cover entity mock."""
+        return mock_feature(
+            "covers",
+            blebox_uniapi.cover.Cover,
+            unique_id="BleBox-shutterBox-2bee34e750b8-position",
+            full_name="shutterBox-position",
+            device_class="shutter",
+            current=None,
+            state=None,
+            has_stop=True,
+            is_slider=True,
+        )
+
+    async def test_init(self, hass):
+        """Test cover default state."""
+
+        entity = (await self.async_mock_entities(hass))[0]
+
+        assert entity.name == "shutterBox-position"
+        assert entity.unique_id == "BleBox-shutterBox-2bee34e750b8-position"
+
+        assert entity.device_class == DEVICE_CLASS_SHUTTER
+
+        assert entity.supported_features & SUPPORT_OPEN
+        assert entity.supported_features & SUPPORT_CLOSE
+        assert entity.supported_features & SUPPORT_STOP
+
+        assert entity.supported_features & SUPPORT_SET_POSITION
+        assert entity.current_cover_position is None
+
+        self.assert_state(entity, None)
+
+    async def test_update(self, hass):
+        """Test cover updating."""
+
+        self.updateable_feature_mock()
+
+        entity = await self.async_updated_entity(hass, 0)
+
+        assert entity.current_cover_position == 22  # 100 - 78
+        self.assert_state(entity, STATE_OPEN)
+
+    async def test_set_position(self, hass):
+        """Test cover position setting."""
+        self.closed_to_position_almost_closed_feature_mock()
+
+        entity = await self.async_updated_entity(hass, 0)
+
+        self.assert_state(entity, STATE_CLOSED)
+        await entity.async_set_cover_position(**{ATTR_POSITION: 1})  # almost closed
+        self.assert_state(entity, STATE_OPENING)
+
+    def unkown_position_feature_mock(self):
+        """Return mocked feature representing an invalid+possible position."""
+        feature_mock = self._feature_mock
+
+        def update():
+            feature_mock.state = 4  # opening
+            feature_mock.current = -1
+
+        feature_mock.async_update = CoroutineMock(side_effect=update)
+
+    async def test_unknown_position(self, hass):
+        """Test cover position setting."""
+        self.unkown_position_feature_mock()
+
+        with patch("homeassistant.components.blebox.cover._LOGGER.warning") as warn:
+            entity = await self.async_updated_entity(hass, 0)
+            self.assert_state(entity, STATE_OPEN)
+            assert entity.current_cover_position is None
+            warn.assert_has_calls(
+                [
+                    call(
+                        "Position for %s is unknown. Try calibrating the device.",
+                        "shutterBox-position",
+                    )
+                ]
+            )
+
+
+class TestGateBox(CoverTest):
```

Comment:
New tests need to be standalone pytest test functions (no classes).

Reply:
Fixed, along with refactoring using pytest fixture parameters.

However, I reuse some tests and helpers (defined in `DefaultBoxTest`) in other platforms (not included in this PR).

So for now, I'm calling those shared tests directly from `DefaultBoxTest` in case there are more changes needed in `test_cover.py`. (Since I'd have to update other platforms not included in this PR).

I'm not sure how to cleanly implement shared tests without classes, though.

Reply:
Your mocks can be classes, its just the tests themselves that need to be standalone pytest test functions